src/pfs/server/api_server.go

```diff
@@ -77,11 +77,28 @@ func (a *apiServer) InspectRepo(ctx context.Context, request *pfs.InspectRepoReq
 	a.versionLock.RLock()
 	defer a.versionLock.RUnlock()
 	ctx = versionToContext(a.version, ctx)
-	clientConn, err := a.getClientConn(a.version)
+
+	clientConns, err := a.router.GetAllClientConns(a.version)
 	if err != nil {
 		return nil, err
 	}
-	return pfs.NewInternalAPIClient(clientConn).InspectRepo(ctx, request)
+
+	var repoInfos []*pfs.RepoInfo
+	for _, clientConn := range clientConns {
+		repoInfo, err := pfs.NewInternalAPIClient(clientConn).InspectRepo(ctx, request)
+		if err != nil {
+			return nil, err
+		}
+		repoInfos = append(repoInfos, repoInfo)
+	}
```

Comment:
This can be done concurrently with a WaitGroup and a channel to collect the results.